Frameworks/include/MockClass.h

```diff
@@ -0,0 +1,824 @@
+
+// Mock a three-arg internal abstract class method in high level C++/CX
+#define CX_MOCK_INTERNAL_ABSTRACT_METHOD_3(INTERFACE_NAME, METHOD_NAME) \
+    CX_BUILD_MOCK_METHOD_IMPL(INTERFACE_NAME, METHOD_NAME, 3, ABSTRACT_CLASS, internal)
+
+// Mock a four-arg internal abstract class method in high level C++/CX
+#define CX_MOCK_INTERNAL_ABSTRACT_METHOD_4(INTERFACE_NAME, METHOD_NAME) \
+    CX_BUILD_MOCK_METHOD_IMPL(INTERFACE_NAME, METHOD_NAME, 4, ABSTRACT_CLASS, internal)
+
+// Mock a five-arg internal abstract class method in high level C++/CX
+#define CX_MOCK_INTERNAL_ABSTRACT_METHOD_5(INTERFACE_NAME, METHOD_NAME) \
+    CX_BUILD_MOCK_METHOD_IMPL(INTERFACE_NAME, METHOD_NAME, 5, ABSTRACT_CLASS, internal)
+
+// Mock a six-arg internal abstract class method in high level C++/CX
+#define CX_MOCK_INTERNAL_ABSTRACT_METHOD_6(INTERFACE_NAME, METHOD_NAME) \
+    CX_BUILD_MOCK_METHOD_IMPL(INTERFACE_NAME, METHOD_NAME, 6, ABSTRACT_CLASS, internal)
+
+// Mock a seven-arg internal abstract class method in high level C++/CX
+#define CX_MOCK_INTERNAL_ABSTRACT_METHOD_7(INTERFACE_NAME, METHOD_NAME) \
+    CX_BUILD_MOCK_METHOD_IMPL(INTERFACE_NAME, METHOD_NAME, 7, ABSTRACT_CLASS, internal)
+
+// Mock an eight-arg internal abstract class method in high level C++/CX
+#define CX_MOCK_INTERNAL_ABSTRACT_METHOD_8(INTERFACE_NAME, METHOD_NAME) \
+    CX_BUILD_MOCK_METHOD_IMPL(INTERFACE_NAME, METHOD_NAME, 8, ABSTRACT_CLASS, internal)
+
+// Mock a nine-arg internal abstract class method in high level C++/CX
+#define CX_MOCK_INTERNAL_ABSTRACT_METHOD_9(INTERFACE_NAME, METHOD_NAME) \
+    CX_BUILD_MOCK_METHOD_IMPL(INTERFACE_NAME, METHOD_NAME, 9, ABSTRACT_CLASS, internal)
+
+// Mock a ten-arg internal abstract class method in high level C++/CX
+#define CX_MOCK_INTERNAL_ABSTRACT_METHOD_10(INTERFACE_NAME, METHOD_NAME) \
+    CX_BUILD_MOCK_METHOD_IMPL(INTERFACE_NAME, METHOD_NAME, 10, ABSTRACT_CLASS, internal)
+
+// Mock an internal abstract read/write property getter/setter
+#define CX_MOCK_INTERNAL_ABSTRACT_PROPERTY(INTERFACE_NAME, PROPERTY_NAME) \
+    CX_BUILD_MOCK_PROPERTY_IMPL(INTERFACE_NAME, PROPERTY_NAME, ABSTRACT_CLASS, internal)
+
+// Mock an internal abstract readonly property getter
+#define CX_MOCK_INTERNAL_ABSTRACT_READONLY_PROPERTY(INTERFACE_NAME, PROPERTY_NAME) \
+    CX_BUILD_MOCK_READONLY_PROPERTY_IMPL(INTERFACE_NAME, PROPERTY_NAME, ABSTRACT_CLASS, internal)
+
+//*********************************************************************************************************************
+// DO NOT directly call any of the private helper macros/functions below.
+//*********************************************************************************************************************
+
+/////////////////////////////////////////////////////////////
+// Validates that a callback has been set before invoking it.
+/////////////////////////////////////////////////////////////
+#if !defined CUSTOM_MOCK_CALLBACK_VALIDATOR
+namespace Arcadia {
```

Comment:
Sounds pretty good except the mocks can be used in unit tests too.... 

---
In reply to: [68810797](https://github.com/Microsoft/WinObjC/pull/592#discussion_r68810797) [](ancestors = 68810797,68809195)